devices/device.py

```diff
@@ -40,55 +40,98 @@ def _create_device(self, device_data, message):
         )
 
     def get_numeric_value(self, value, device):
         Domoticz.Error(
             'Device with alias "' + self.alias + '" for key ' +
-            self.value_key + ' can not calculate numeric value'
+            "_".join(self.value_keys) + ' can not calculate numeric value'
         )
 
     def get_string_value(self, value, device):
         Domoticz.Error(
             'Device with alias "' + self.alias + '" for key ' +
-            self.value_key + ' can not calculate string value'
+            "_".join(self.value_keys) + ' can not calculate string value'
         )
 
+    def get_sn_values(self, key, value, device):
+        return (self.get_string_value(value, device),self.get_numeric_value(value, device))
+
     def handle_message(self, device_data, message):
         device_address = device_data['ieee_addr']
         device = self.get_device(device_address, self.alias)
+        n_value = None
+        s_value = None
+        color_value = None
+        Domoticz.Debug("handle_message for device: '"+str(device)+"', alias: '"+self.alias+"' with # items: '"+str(len(message.raw))+"'")
+        for item in message.raw:
+            Domoticz.Debug("handle_message the message item: '"+item+"' message: '"+str(message.raw[item])+"'")
 
         if (device == None):
             # Due to internal domoticz bug, app crashes if we try to use device just after we create it
             # so just create and exit for now
             # device = self._create_device(device_data, message)
             return self._create_device(device_data, message)
 
-        if (self.value_key not in message.raw):
-            # There is no way to properly handle heartbeat messages as nValue and sValue are mandatory for device update
-            Domoticz.Debug('Received heartbeat message from device "' + device.Name + '"')
-            return None
+        for key in self.value_keys:
+            if (key not in message.raw):
+                # There is no way to properly handle heartbeat messages as nValue and sValue are mandatory for device update
+                Domoticz.Debug('Received heartbeat message from device "' + device.Name + '"')
+                return None
 
-        value = message.raw[self.value_key]
-        n_value = self.get_numeric_value(value, device)
-        s_value = self.get_string_value(value, device)
+            if (key != 'color'):
+                value = message.raw[key]
+                (s_value, n_value) = self.get_sn_values(key, value, device)
+                
+
+        # value = message.raw[self.value_key]
+        # n_value = self.get_numeric_value(value, device)
+        # s_value = self.get_string_value(value, device)
 
         signal_level = message.get_signal_level()
         battery_level = message.get_battery_level()
-
+        
+        if ("state" in message.raw and message.raw['state'].upper()=='OFF'):
+            s_value = 'Off'
+            n_value = 0
+
+
+        if ("color" in message.raw):
```

Comment:
What do you think about moving this part to `ColorLight` as it doesn't seem required for all other devices